spring-integration-core/src/main/java/org/springframework/integration/mapping/support/JsonHeaders.java

```diff
@@ -103,9 +105,9 @@ else if (classValue != null) {
 	 * @return the {@link ResolvableType} based on provided class components
 	 * @since 5.2.4
 	 */
-	public static ResolvableType buildResolvableType(Class<?> targetClass, @Nullable Class<?> contentClass,
+	public static ResolvableType buildResolvableType(@Nullable Class<?> targetClass, @Nullable Class<?> contentClass,
 			@Nullable Class<?> keyClass) {
-
+		Assert.state(targetClass != null, "The targetClass must not be null");
```

Comment:
This is not correct.
The `targetClass` must not be null.

Consider to mark `getClassForValue()` with a `@Contract` we talked yesterday in other place:
```
@Contract("_, null -> null; _, !null -> !null")
```